### PR TITLE
Case insensitive matching of field names

### PR DESCRIPTION
**Pull request by ychen-bloxer**
there is an issue returning the values for an object when requesting the list of DNS Server Groups (API object auth_nsg) via API when using the fields.

When there is no specific filter/tag applied in the query, everything looks good and the complete data is displayed.

### CODE REVIEW

**Comment by chinmayb** [comment]
Please give some more details of the fix and please avoid using internal tickets.

File: gorm/fields.go [hunk 1]
```diff
@@ -76,35 +76,53 @@ fields:
...
+		for i := 0; i < objType.NumField(); i++ {
+			if strings.EqualFold(name, strings.ToLower(strings.ReplaceAll(queryFieldName, "_", ""))) {
+				return true
+			}
+		}
+		return false
+	})
+
 	if !ok {
+		fmt.Printf("===> no field found")
```
Comment (daniel-garcia) [inline gorm/fields.go:92]: Use log package

**Review by daniel-garcia — CHANGES_REQUESTED** [review]

**Comment by ychen-bloxer** [comment]
link issue https://infoblox.atlassian.net/browse/ATLAS-13969

File: gorm/fields.go [hunk 2]
```diff
@@ -3,37 +3,39 @@ package gorm
 import (
 	"context"
 	"fmt"
+	"github.com/infobloxopen/atlas-app-toolkit/util"
```
Comment (brocwoodworthIBLX) [inline gorm/fields.go:6]: please revert the change in import order. usually you can run `go fmt ./...` in the parent directory and it will sort these for you

File: gorm/fields.go [hunk 3]
```diff
@@ -75,36 +77,55 @@ fields:
 	return toPreload, nil
 }
 
-func handlePreloads(f *query.Field, objType reflect.Type) ([]string, error) {
-	sf, ok := objType.FieldByName(util.Camel(f.GetName()))
+func handlePreloads(f *query.Field, objType reflect.Type, ignoreCase ...bool) ([]string, error) {
+	queryFieldName := f.GetName()
+	fmt.Printf("Query name = %v\n", queryFieldName)
```
Comment (brocwoodworthIBLX) [inline gorm/fields.go:82]: use log package here instead of fmt

File: gorm/fields.go [hunk 4]
```diff
@@ -75,36 +77,55 @@ fields:
...
+
+	var sf reflect.StructField
+	var ok bool
+	if len(ignoreCase) > 0 && ignoreCase[0] {
```
Comment (brocwoodworthIBLX) [inline gorm/fields.go:86]: maybe I'm missing something but why is `ignoreCase` a `...bool` instead of just a plain old `bool`?

**Review by brocwoodworthIBLX — COMMENTED** [review]
couple of minor things and a question to be addressed

File: gorm/fields.go (hunk 4 above)
Comment (ychen-bloxer) [reply gorm/fields.go:86]: This is an "optional" parameter for backward compatibility.

Comment (brocwoodworthIBLX) [reply gorm/fields.go:86]: ok that makes sense. if that's the case then make sure that there is a unti test for `ignoreCase` being `false`, and one for there being no `ignoreCase` value at all

Comment (ychen-bloxer) [reply gorm/fields.go:86]: Yes, there are. In test file, the function has been called as followed
	FieldSelectionStringToGorm(context.Background(), test.fs, &Model{}) // no "ignoreCase" = false
	FieldSelectionStringToGorm(context.Background(), test.fs, &Model{}, true)

Comment (brocwoodworthIBLX) [reply gorm/fields.go:86]: it might be nice to have at least one `FieldSelectionStringToGorm(context.Background(), test.fs, &Model{}, false)` just for completion's sake

**Review by brocwoodworthIBLX — APPROVED** [review]
LGTM

File: gorm/collection_operators.go [hunk 5]
```diff
@@ -16,7 +16,7 @@ type SortingCriteriaConverter interface {
 }
 
 type FieldSelectionConverter interface {
-	FieldSelectionToGorm(ctx context.Context, fs *query.FieldSelection, obj interface{}) ([]string, error)
+	FieldSelectionToGorm(ctx context.Context, fs *query.FieldSelection, obj interface{}, ignoreCase ...bool) ([]string, error)
```
Comment (daniel-garcia) [inline gorm/collection_operators.go:19]: this needs to be a option struct{} not a boolean. consider the meaning is for multiple options.

**Review by daniel-garcia — CHANGES_REQUESTED** [review]

**Review by daniel-garcia — APPROVED** [review]
LGTM.
